src/utils/retry.rs

```diff
@@ -0,0 +1,821 @@
+use rand::Rng;
```

Comment:
I used my own retry logic in the past for monitors, but decided to switch to `reqwest-retry` crate which has most of these capabilities built-in, like exponential backoff, retry strategies (e.g. only retry on certain error codes).

Not a blocker, but I would recommend checking it out.